src/components/AgmExtension/ExtensionRequest.vue

```diff
@@ -268,61 +266,136 @@ export default class ExtensionRequest extends Vue {
+    else if (
+      this.isFirstAgm &&
+      this.data.isPrevExtension &&
+      this.extensionExpiryDateText &&
+      this.data.incorporationDate
+    ) {
       const cutOffYyyyMmDd = DateUtilities.addMonthsToDate(
         30, DateUtilities.dateToYyyyMmDd(this.data.incorporationDate))
       const cutOffDate = DateUtilities.yyyyMmDdToDate(cutOffYyyyMmDd)
       const expiryDate = DateUtilities.yyyyMmDdToDate(this.extensionExpiryDateText)
-      // IF ExpirationDate >= (IncorporationDate + 18 months + 12 months) --> INELIGIBLE
-      // ELSE --> Check ELSE BLOCK
+
+      // IF ExpirationDate >= (IncorporationDate + 18 months + 12 months)
+      // THEN NOT ELIGIBLE
       if ((DateUtilities.daysBetweenTwoDates(cutOffDate, expiryDate)) >= 0) {
         this.data.isEligible = false
-      } else {
-        // IF CurrentDate > (ExpirationDate + 5 days) --> INELIGIBLE
-        // ELSE --> ELIGIBLE --> ExtensionDuration = MIN(6, 12 - totalExtensionApproved)
-        // totalExtensionApproved = (ExpirationDate - 18 months - IncorporationDate) in months
-        const currentDate = DateUtilities.yyyyMmDdToDate(this.getCurrentDate)
-        if ((DateUtilities.daysBetweenTwoDates(expiryDate, currentDate)) > 5) {
-          this.data.isEligible = false
-        } else {
-          this.data.isEligible = true
-          const totalExtensionApproved = DateUtilities.subtractDates(
-            DateUtilities.dateToYyyyMmDd(this.data.incorporationDate), this.extensionExpiryDateText) - 18
-          this.data.extensionDuration = Math.min(6, (12 - totalExtensionApproved))
-        }
+      }
+      // ELSE IF CurrentDate > (ExpirationDate + 5 days)
+      // THEN NOT ELIGIBLE
+      else if ((DateUtilities.daysBetweenTwoDates(expiryDate, currentDate)) > 5) {
+        this.data.isEligible = false
+      }
+      // ELSE ELIGIBLE, ExtensionDuration = MIN(6, 12 - totalExtensionApproved)
+      else {
+        this.data.isEligible = true
+        // totalExtensionApproved: (ExpirationDate - 18 months - IncorporationDate) in months
+        const totalExtensionApproved = DateUtilities.subtractDates(
+          DateUtilities.dateToYyyyMmDd(this.data.incorporationDate), this.extensionExpiryDateText) - 18
+        this.data.extensionDuration = Math.min(6, (12 - totalExtensionApproved))
+      }
+    }
+
+    // ELSE IF NOT first agm extension
+    // AND NOT extension requested this year
+    else if (
+      !this.isFirstAgm &&
+      !this.data.isPrevExtension &&
+      this.previousAgmDateText
+    ) {
+      const cutOffYyyyMmDd = DateUtilities.addMonthsToDate(
+        15, this.previousAgmDateText)
+      const cutOffDate = DateUtilities.yyyyMmDdToDate(cutOffYyyyMmDd)
+
+      // IF CurrentDate > (PrevAgmDate + 15 Months + 5 days)
+      // THEN NOT ELIGIBLE
+      if ((DateUtilities.daysBetweenTwoDates(cutOffDate, currentDate) - 5) > 0) {
+        this.data.isEligible = false
+      }
+      // ELSE ELIGIBLE, ExtensionDuration = 6 months
+      else {
```

Comment:
Leo, I don't see this else statement in Mihai's notes. Did you confirm this by any chance with him? There might be a special case for the extension duration. The else here is probably correct I think but just in case!
![rule 2](https://github.com/bcgov/business-filings-ui/assets/122301442/27053539-0ff5-4b9f-9ec5-f7a2de827b6e)

Reply:
Hey Karim, I found in the non-final scenarios document that there is an "else" statement (the scenario numbers between the 2 docs don't match but this is for `isFirstAgm == false && isPrevExtension == false`. I'll check in with Mihai to be sure however and will update in this thread!

<img width="666" alt="Screenshot 2023-11-14 at 11 56 17 AM" src="https://github.com/bcgov/business-filings-ui/assets/122323255/902df321-1f5b-4a22-bdbc-c3039d5fefd7">

Reply:
Just checked with Mihai and the else block is correct, we want to grant the extension on the basis of `isFirstAgm == false && isPrevExtension == false` and it is less than the time constraint. 

It was just missing from the final scenarios document. It was actually in the outdated version and was not copied over the the final version that you have!

Reply:
Awesome, thanks a lot Leo! Resolving.